spec/models/price_policy_spec.rb

```diff
@@ -34,6 +34,29 @@
     end
   end
 
+  context "current and newest" do
+    let(:price_policy) { FactoryGirl.create(:instrument_price_policy) }
+    let(:overlapping_price_policy) do
+      FactoryGirl.create(
+        :instrument_price_policy,
+        start_date: (price_policy.start_date + 1.day).beginning_of_day,
+        product: price_policy.product,
+        price_group: price_policy.price_group,
+      )
+    end
+
+    before { overlapping_price_policy.update_attributes(start_date: 1.day.ago.beginning_of_day) }
+
+    it "saves updated price policy" do
+      expect(overlapping_price_policy).to be_valid
+    end
+
+    it "has two overlapping policies" do
```

Comment:
I think it might be smart to do a spec at a higher level, where we have two policies: and older one with a cheaper price and a newer one. Then on the order detail, find the do `assign_price_policy`, which will call `cheapest_price_policy`  and make sure it chooses the correct policy.